library/spec/Ivoz/Provider/Domain/Model/Calendar/CalendarSpec.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace spec\Ivoz\Provider\Domain\Model\Calendar;
+
+use Ivoz\Provider\Domain\Model\Calendar\Calendar;
+use Ivoz\Provider\Domain\Model\Calendar\CalendarDto;
+use Ivoz\Provider\Domain\Model\HolidayDate\HolidayDateInterface;
+use Ivoz\Provider\Domain\Model\Company\CompanyInterface;
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use spec\HelperTrait;
+use Doctrine\Common\Collections\ArrayCollection;
+
+class CalendarSpec extends ObjectBehavior
+{
+    use HelperTrait;
+
+    protected $dto;
+
+    protected $company;
+    protected $holidayDate;
+    protected $holidayDate2;
+
+    function let(
+        CompanyInterface $company,
+        HolidayDateInterface $holidayDate,
+        HolidayDateInterface $holidayDate2
+    ) {
+        $this->dto = $dto = new CalendarDto();
+        $dto->setName('Calendar');
+
+        $this->company = $company;
+        $this->holidayDate = $holidayDate;
+
+        $this
+            ->holidayDate
+            ->getId()
+            ->willReturn(1);
+
+        $this->holidayDate
+            ->setCalendar(Argument::type(Calendar::class))
+            ->willReturn();
+
+        $this
+            ->holidayDate
+            ->checkEventOnTime(Argument::any())
+            ->willReturn(true);
+
+        $this->holidayDate2 = $holidayDate2;
+        $this->holidayDate2
+            ->getId()
+            ->willReturn(2);
+
+        $this->holidayDate2
+            ->setCalendar(Argument::type(Calendar::class))
+            ->willReturn();
+
+        $this
+            ->holidayDate2
+            ->checkEventOnTime(Argument::any())
+            ->willReturn(true);
+
+        $holidayDates = [
+            $holidayDate->getWrappedObject(),
+            $holidayDate2->getWrappedObject()
+        ];
+        $arrayCollection = new ArrayCollection($holidayDates);
+
+        $this->hydrate(
+            $dto,
+            [
+                'company' => $company->getWrappedObject(),
+                'holidayDates' => $arrayCollection
+            ]
+        );
+
+        $this->beConstructedThrough(
+            'fromDto',
+            [$dto]
+        );
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType(Calendar::class);
+    }
+
+    function it_tells_whether_given_day_is_a_holiday()
+    {
+        $this->getterProphecy(
+            $this->holidayDate,
+            [
+                'getEventDate' => (new \DateTime('2019-01-01'))
+            ],
+            true
+        );
+
+        $this->getterProphecy(
+            $this->holidayDate2,
+            [
+                'getEventDate' => (new \DateTime('2019-01-02'))
+            ],
+            true
+        );
+
+        $this
+            ->isHolidayDate(new \DateTime('2019-01-02'))
+            ->shouldReturn(true);
+
+        $this
+            ->isHolidayDate(new \DateTime('2019-01-02'))
```

Comment:
Duplicate test.

Instead of updating the date, you could test the same date in different timezone :)

Reply:
Done :)